### ISSUE TITLE
Train and test cost

### ISSUE THREAD

**Issue by yahooo-m** [showlab/VideoLISA#5]
Hi, sorry to bother u again. 
I’d like to know the training and testing times. I’ve noticed that testing takes an especially long time. For example, using two A100 GPUs, testing Mevis required 24 hours.

**Comment by JosephPai** [comment]
Hi @yahooo-m, sorry for the late reply.
Typically, we use 8 GPUs for fast inference. When you try to use 2 GPUs, please remember to set `SUBSET_NUM=2` in the `run_inference_mevis.sh` script.
I have tried to reproduce your case as using 2 A10 GPUs. The estimated time is 3~4 hours.
A100 GPUs should be faster than A10 GPUs in theory.

<img width="1776" alt="image" src="https://github.com/user-attachments/assets/c5fd56bd-d4dc-44bf-a1e1-ca52d7b7c025" />

I may suggest first check your transformer version. Different transformer version, especially old version, can slow down the process. Also ensure you really load the model into GPUs.
After that, you may try to break down the inference time to three parts, 1) data loading and saving, 2) inference of MLLM, 3) inference of SAM, to see where is the bottleneck.

**Comment by Lexarymade** [comment]
Hey, dear author, I've also encountered the above problem in H100. So, I checked the evaluation process following your instructions, and that's what I found:
- load image time cost:  13.122689247131348
- model time cost:  26.453353881835938
- post optimize time cost:  155.03177499771118
Also, I encountered the flash-attention problem while the mllm was inferring: You are not running the flash-attention implementation, expect numerical differences. The transformer version is the same.

**Comment by JosephPai** [comment]
Hi @Lexarymade , thanks for the info.
I also have the warning regarding flash-attn during inference. It does not affect the inference speed on my side.
I will try to find an H100 GPU to test the latency.

Btw, what is the `post optimize time cost` in your test?

**Comment by Lexarymade** [comment]
maybe i mentioned wrong, the `post optimize time cost` refers to the code executed time after the mllm  output

**Comment by yahooo-m** [comment]
@Lexarymade @JosephPai I also checked my inference time and changed different versions of torch and flash-attn. The inference speed is still slow.  Have u fix it? @Lexarymade

**Comment by JosephPai** [comment]
Hi @yahooo-m, I tried to run the evaluation using another server with 8 L40S (48G) GPUs. Even without turning on flash attention, the evaluation can finish within 1 hour. When reducing the number of GPUs to 2, the evaluation is estimated to finish with about 4 hours.
Unfortunately I am not able to reproduce your issue on my side. Maybe you need to check the time usage for each component, including 1) loading video time, 2) MLLM inference time (only one time for each video), 3) SAM inference time (depending on number of frames), 4) saving results (depending on number of frames). If you have successfully put the model on GPU, the inference of MLLM and SAM should be fast.
I highly suspect the data IO part, as some videos may contain a lot of frames.

**Comment by JosephPai** [comment]
@Lexarymade , yes you are correct. Ref-DAVIS datasets is annotated by four annotators. The final metric number is the average of them.
